### ISSUE TITLE
Host alerts are not logged and host notification will be sent at the first soft state

### ISSUE THREAD

**Issue by ccztux** [NagiosEnterprises/nagioscore#484]
On my test system, build from github master branch, I have found the issue.

Content of nagios.log:
```
[1518953116] Nagios 4.4.0 starting... (PID=5241)
[1518953116] Local time is Sun Feb 18 12:25:16 CET 2018
[1518953116] LOG VERSION: 2.0
[1518953116] qh: Socket '/usr/local/nagios/var/rw/nagios.qh' successfully initialized
[1518953116] qh: core query handler registered
[1518953116] wproc: Successfully registered manager as @wproc with query handler
[1518953116] Caught SIGCHLD, calling waitpid() with WNOHANG|WUNTRACED
[1518953116] wproc: Registry request: name=Core Worker 5892;pid=5892
[1518953116] wproc: Registry request: name=Core Worker 5894;pid=5894
[1518953116] wproc: Registry request: name=Core Worker 5893;pid=5893
[1518953116] wproc: Registry request: name=Core Worker 5891;pid=5891
[1518953159] SERVICE ALERT: localhost;HTTP;CRITICAL;SOFT;1;CRITICAL - Socket timeout
[1518953174] SERVICE ALERT: localhost;PING;CRITICAL;SOFT;1;CRITICAL - Plugin timed out
[1518953174] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
[1518953189] wproc: Core Worker 5891: job 0 (pid=5903) timed out. Killing it
[1518953189] wproc: CHECK job 0 from worker Core Worker 5891 timed out after 30.04s
[1518953189] wproc:   host=localhost; service=(null);
[1518953189] wproc:   early_timeout=1; exited_ok=0; wait_status=0; error_code=62;
[1518953189] Warning: Check of host 'localhost' timed out after 30.04 seconds
[1518953189] HOST NOTIFICATION: nagiosadmin;localhost;DOWN;notify-host-by-email;(Host check timed out after 30.04 seconds)
[1518953189] wproc: Core Worker 5891: job 0 (pid=5903): Dormant child reaped
[1518953202] SERVICE ALERT: localhost;SSH;CRITICAL;HARD;1;CRITICAL - Socket timeout
[1518953234] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
[1518953279] SERVICE ALERT: localhost;HTTP;CRITICAL;HARD;2;CRITICAL - Socket timeout
[1518953294] SERVICE ALERT: localhost;PING;CRITICAL;HARD;2;CRITICAL - Plugin timed out
[1518953294] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
[1518953354] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
[1518953414] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
[1518953474] Warning: Return code of 25 for check of service 'modpd status' on host 'localhost' was out of bounds.
```

This looks also really strange (/nagios/cgi-bin/extinfo.cgi?type=1&host=localhost);

```
Current Attempt:	2/3  (HARD state)
```

**Comment by ccztux** [comment]
Host UP notificaton was sent twice (notification_interval set to 0):
```
[1518955960] HOST NOTIFICATION: nagiosadmin;localhost;DOWN;notify-host-by-email;(Host check timed out after 30.04 seconds)
[1518956144] HOST NOTIFICATION: nagiosadmin;localhost;UP;notify-host-by-email;PING OK - Packet loss = 0%, RTA = 0.06 ms
[1518956207] HOST NOTIFICATION: nagiosadmin;localhost;UP;notify-host-by-email;PING OK - Packet loss = 0%, RTA = 0.06 ms
```

**Comment by hedenface** [comment]
Are you seeing any issues with service notifications going out as well?

**Comment by hedenface** [comment]
I'm seeing a double up on the check count as well for hosts, but unable to duplicate for services. Is reproducible regardless. I'm sure this was due to my cleanup of the check logic code a while ago.

**Comment by jomann09** [comment]
This has been fixed, and in 4.4.3 I tested it and is working fine. Please open another issue if there is anything else weird with this.
